app/models/nic/base.rb

```diff
@@ -148,6 +151,13 @@ def host_managed?
 
     protected
 
+    def compute_resource_requires_fields
+      if host.compute_resource.present?
```

Comment:
I believe it's still misplaced, since the issue is lack of validation on hosts.  That it goes on to affect NICs doesn't mean the validation should be here.